client/app/components/land-use-action.js

```diff
@@ -74,12 +74,16 @@ export default class LandUseActionComponent extends Component {
     return this.allLandUseActionOptions.filter((option) => pasForm[option.field] || pasForm[option.attr1] || pasForm[option.attr2]);
   }
 
-  // Actions selected by user in dropdown, or already
-  // existing in CRM
+  // Actions selected by user in dropdown ("new"), or already
+  // existing in CRM ("existing")
   get landUseActionSelections() {
+    const { pasForm } = this.args;
+    // Filter the "new" actions array so that the action does not exist on BOTH the "exisiting" and "new" arrays,
+    // when user enters a value in the action inputs (and therefore updates the model)
+    const filteredNewActions = this.newLandUseActionSelections.filter((action) => !pasForm[action.field] && !pasForm[action.attr1] && !pasForm[action.attr2]);
```

Comment:
@trbmcginnis could we just filter based on the `action.field`?  Just trying to understand why we need to explicitly look for `attr1` and `attr2` as well.
```js
    const filteredNewActions = this.newLandUseActionSelections.filter(
      action => !pasForm[action.field]
    );
```

Reply:
If you look at how "exisiting" actions are defined, they are defined as having at least ONE of their attributes (including `field`, `attr1`, `attr2`) saved to the model (I might actually rename these to make them more clear). So if an action has ANY of their fields stored on the model/db then they will be put into the existing actions. This is to account for the case when a user does a partial save after having filled out only some inputs. 

existing = `this.allLandUseActionOptions.filter((option) => pasForm[option.field] || pasForm[option.attr1] || pasForm[option.attr2]);` 

So if we're trying to include only "new" actions in the `filteredNewActions` array (and make sure they don't exist in "existing") we need to make sure that the action has NO attributes saved at all on the model. They're basically just local variables at that point.

Reply:
From the action where users add a new selection, it looks like `field` is the the unique key for identifying / adding the new land use action:
```js
 @action
  addNewLandUseActionSelection() {
    const selection = this.selectedLandUseAction.field;

    if (selection) {
      this.newLandUseActionSelections.pushObject(this.selectedLandUseAction);
      this.selectedLandUseAction = {};
    }
  }
```

But if I'm understanding, it is possible for us to get existing land use actions from CRM where the field is empty but the `attrs` are not?  In other words, `attrs` don't require the `field`?

Reply:
I think the `const selection = this.selectedLandUseAction.field` is there because if you just did `this.selectedLandUseAction`, whether it was an empty object or a filled object, it would still come back as truthy. 

There are technically "requirements" for what you can submit to CRM but that's handled on the Submit button's validations (if an action has extra attributes and the count field is filled in, then the extra attributes are required to be filled in to Submit). But because a user can hit `save` at any point in the process, there might be times with any of those fields (`field`, `attr1`, or `attr2`) are filled and none of the others are.